src/lib/history/history-cleanup-processor.js

```diff
@@ -16,20 +16,89 @@ class HistoryCleanupProcessor {
     {
         this._logger.info("[_onDbConnected] ...");
         this._latestSnapshot = null;
+
+        this._registerStatements()
+    }
+
+    _registerStatements()
+    {
+        this._registerStatement('FIND_OLDEST_SNAPSHOT_ID', 'SELECT `id` FROM `snapshots` WHERE `date` < ? ORDER BY `date` LIMIT 1')
+
+        this._registerStatement('FIND_DIFFS_FOR_SNAPSHOT', 'SELECT `id` FROM `diffs` WHERE `snapshot_id` = ?')
+
+        this._registerStatement('DELETE_DIFF_ITEMS_FOR_DIFF', 'DELETE FROM `diff_items` WHERE `diff_id` = ?')
+
+        this._registerStatement('DELETE_DIFFS_FOR_SNAPSHOT', 'DELETE FROM `diffs` WHERE `snapshot_id` = ?')
+
+        this._registerStatement('DELETE_SNAP_ITEMS_BY_SNAPSHOT_ID', 'DELETE FROM `snap_items` WHERE `snapshot_id` = ?')
+
+        this._registerStatement('DELETE_SNAPSHOT_BY_ID', 'DELETE FROM `snapshots` WHERE `id` = ?')
+    }
+
+    _retryToDelete()
+    {
+        return this.cleanDb()
     }
 
     cleanDb()
     {
-        const snapDate = [new Date(Date.now() - this._days * 24 * 60 * 60 * 1000)]
+        const snapDate = [moment().subtract(this._days, 'days').format()];
+
+        return Promise.resolve()
+            .then(() => this.fetchSnapshot(snapDate))
+            .then(snapshot => {
+                return Promise.resolve(snapshot.map(item => this.fetchDiffs(item.id)))
+                    .then(() => {
+                        return this.deleteSnapshot(snapshot[0].id)
+                    })
+            })
+    }
+
+    fetchSnapshot(date)
+    {
+        return this._execute('FIND_OLDEST_SNAPSHOT_ID', date)
+    }
 
-        Promise.resolve()
-            .then(() => this._dbAccessor.fetchCleanedSnapshots(snapDate))
-            .then(snapshotIds => {
-                return this._dbAccessor.deleteDiffs(snapshotIds)
-                    .then(() => snapshotIds)
+    fetchDiffs(snapshotId)
+    {
+        return this._execute('FIND_DIFFS_FOR_SNAPSHOT', [snapshotId])
+            .then(diffs => {
+                return Promise.resolve(diffs.map(item => {
```

Comment:
i'd use this instead:
```
return Promise.serial(diffs, item => {
```
The result would be the same, but it is shorter, easier to read. Also can replace with Promise.parallel in future to run things in parallel.